File: progra3-0583-equipo5-frontend/Frontend_C#/SoftPac/SoftPacBusiness/CuentasAcreedorBO.cs
using SoftPac.Model;
using SoftPac.Persistance.DAO;
using System.Collections.Generic;

namespace SoftPac.Business
{
    public class CuentasAcreedorBO
    {
        private CuentasAcreedorDAOImpl cuentasAcreedorDAO;

        public CuentasAcreedorBO()
        {
            this.cuentasAcreedorDAO = new CuentasAcreedorDAOImpl();
        }

        public IList<CuentasAcreedorDTO> obtenerPorAcreedor(int acreedor_id)
        {
            return this.cuentasAcreedorDAO.ObtenerPorAcreedor(acreedor_id);
        }

        public IList<CuentasAcreedorDTO> ListarTodos()
        {
            return this.cuentasAcreedorDAO.ListarTodos();
        }

        public IList<CuentasAcreedorDTO> ObtenerPorAcreedor(int acreedor_id)
        {
            return this.obtenerPorAcreedor(acreedor_id);
        }

        public CuentasAcreedorDTO ObtenerPorId(int cuenta_bancaria_id)
        {
            return this.cuentasAcreedorDAO.ObtenerPorId(cuenta_bancaria_id);
        }

        public int insertar(CuentasAcreedorDTO cuenta)
        {
            return this.cuentasAcreedorDAO.Insertar(cuenta);
        }

        public int modificar(CuentasAcreedorDTO cuenta)
        {
            return this.cuentasAcreedorDAO.Modificar(cuenta);
        }

        public int eliminarLogico(int cuenta_bancaria_id, UsuariosDTO usuario)
        {
            var dto = new CuentasAcreedorDTO
            {
                CuentaBancariaId = cuenta_bancaria_id,
                UsuarioEliminacion = usuario,
                FechaEliminacion = System.DateTime.Now
            };
            return this.cuentasAcreedorDAO.eliminarLogico(dto);
        }

        public int eliminarLogico(CuentasAcreedorDTO cuenta)
        {
            return this.cuentasAcreedorDAO.eliminarLogico(cuenta);
        }
    }
}

File: Backend_Java/SoftPacBusiness/src/main/java/pe/edu/pucp/softpac/bo/CuentasAcreedorBO.java
package pe.edu.pucp.softpac.bo;

import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softpac.dao.CuentasAcreedorDAO;
import pe.edu.pucp.softpac.daoImpl.CuentasAcreedorDAOImpl;
import pe.edu.pucp.softpac.model.CuentasAcreedorDTO;
import pe.edu.pucp.softpac.model.UsuariosDTO;

public class CuentasAcreedorBO {
    private CuentasAcreedorDAO cuentaAcreedorDAO;
   
    public CuentasAcreedorBO(){
        cuentaAcreedorDAO = new CuentasAcreedorDAOImpl();
    }
    
     public ArrayList<CuentasAcreedorDTO> obtenerPorAcreedor(Integer acreedor_id){
        if (acreedor_id == null || acreedor_id <= 0) {
            throw new IllegalArgumentException("El ID del acreedor es obligatorio");
        }
        return cuentaAcreedorDAO.obtenerPorAcreedor(acreedor_id);
    }
    
    
    
    public ArrayList<CuentasAcreedorDTO> listarTodos(){
        return (ArrayList<CuentasAcreedorDTO>) cuentaAcreedorDAO.listarTodos();
    }
    
    public CuentasAcreedorDTO obtenerPorId(Integer cuenta_acreedor_id){
        CuentasAcreedorDTO cuentaAcreedorDTO = new CuentasAcreedorDTO();
        cuentaAcreedorDTO.setCuenta_bancaria_id(cuenta_acreedor_id);
        return this.cuentaAcreedorDAO.obtenerPorId(cuenta_acreedor_id);
    }
    
    public Integer insertar(CuentasAcreedorDTO cuenta){
        return this.cuentaAcreedorDAO.insertar(cuenta);
    }
    
    public Integer modificar(CuentasAcreedorDTO cuenta){
        return this.cuentaAcreedorDAO.modificar(cuenta);
    }
    
    public Integer eliminarLogico(CuentasAcreedorDTO cuenta, UsuariosDTO usuarioActual){
        cuenta.setFecha_eliminacion(new Date());
        cuenta.setUsuario_eliminacion(usuarioActual);
        return cuentaAcreedorDAO.eliminarLogico(cuenta);
    }
    
    public Integer eliminarLogico(CuentasAcreedorDTO cuenta){
        
        return cuentaAcreedorDAO.eliminarLogico(cuenta);
    }    
    
}